src/main/java/io/github/enkarin/bookcrossing/refresh/service/RefreshService.java

```diff
@@ -1,27 +1,45 @@
 package io.github.enkarin.bookcrossing.refresh.service;
 
+import io.github.enkarin.bookcrossing.exception.RefreshTokenInvalidException;
+import io.github.enkarin.bookcrossing.exception.TokenNotFoundException;
+import io.github.enkarin.bookcrossing.exception.UserNotFoundException;
 import io.github.enkarin.bookcrossing.refresh.model.Refresh;
 import io.github.enkarin.bookcrossing.refresh.repository.RefreshRepository;
+import io.github.enkarin.bookcrossing.registation.dto.AuthResponse;
+import io.github.enkarin.bookcrossing.security.jwt.JwtProvider;
 import io.github.enkarin.bookcrossing.user.model.User;
 import io.github.enkarin.bookcrossing.user.repository.UserRepository;
 import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
 import java.time.Instant;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.ZoneOffset;
-import java.util.Optional;
 import java.util.UUID;
 
 @Service
 @RequiredArgsConstructor
+@Transactional
 public class RefreshService {
 
     private final RefreshRepository refreshRepository;
     private final UserRepository userRepository;
+    private final JwtProvider jwtProvider;
 
-    public String createToken(final String login) {
+    public AuthResponse createTokens(final String token) {
+        final Refresh refresh = refreshRepository.findById(token).orElseThrow(TokenNotFoundException::new);
+        if (LocalDate.ofEpochDay(refresh.getDate()).isAfter(LocalDate.now())) {
+            return AuthResponse.create(jwtProvider.generateToken(refresh.getUser()),
+                    createAccess(refresh.getUser()));
+        } else {
+            refreshRepository.delete(refresh);
+            throw new RefreshTokenInvalidException();
```

Comment:
You have to create a test for this method. Throwing RefreshTokenInvalidException will lead to transaction rollback thus no data will be deleted from the database

Reply:
Take a look at noRollbackFor https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/transaction/annotation/Transactional.html

Reply:
Added